py/circuitpy_mpconfig.mk

```
@@ -95,6 +95,9 @@ CFLAGS += -DCIRCUITPY_BITBANG_APA102=$(CIRCUITPY_BITBANG_APA102)
 CIRCUITPY_BITBANGIO ?= $(CIRCUITPY_FULL_BUILD)
 CFLAGS += -DCIRCUITPY_BITBANGIO=$(CIRCUITPY_BITBANGIO)
 
+CIRCUITPY_BITMAPTOOLS ?= $(CIRCUITPY_FULL_BUILD)
```

I suspect you'll want to turn off BITMAP tools for all SAMD21 builds in the ports/atmel-samd version of this file even if they have displayio.
```suggestion
CIRCUITPY_BITMAPTOOLS ?= $(CIRCUITPY_FULL_BUILD) && $(CIRCUITPY_DISPLAYIO)
```

---

Make unfortunately doesn't have boolean or arithmetic expressions, so you have to do this test in a more convoluted way, but you can use an existing test for `CIRCUITPY_DISPLAYIO` at line 158:
```make
ifeq ($(CIRCUITPY_DISPLAYIO),1)
CIRCUITPY_BITMAPTOOLS ?= $(CIRCUITPY_FULL_BUILD)
CIRCUITPY_FRAMEBUFFERIO ?= $(CIRCUITPY_FULL_BUILD)
else
CIRCUITPY_BITMAPTOOLS ?= 0
CIRCUITPY_FRAMEBUFFERIO ?= 0
endif
CFLAGS += -DCIRCUITPY_BITMAPTOOLS=$(CIRCUITPY_BITMAPTOOLS)
CFLAGS += -DCIRCUITPY_FRAMEBUFFERIO=$(CIRCUITPY_FRAMEBUFFERIO)

---

Thanks for the comments.  Dan, special thanks for pointing me in the right direction on the conditionals for AND of DISPLAYIO and BITMAPTOOLS.

I updated the configuration files for the other smaller builds that were failing to not include this module.  I'll wait to see if all those builds pass before resolving this comment.

---

Ok after a couple of tries I got all of the `mpconfigboard.mk` files updated.  All the builds pass, so I'm marking this as resolved.